costmap_2d/src/costmap_2d_ros.cpp

```diff
@@ -463,14 +463,16 @@ void Costmap2DROS::mapUpdateLoop(double frequency)
       unsigned int x0, y0, xn, yn;
       layered_costmap_->getBounds(&x0, &xn, &y0, &yn);
       publisher_->updateBounds(x0, xn, y0, yn);
-
       ros::Time now = ros::Time::now();
-      if (last_publish_ + publish_cycle < now)
+      ROS_WARN_COND(now < prev_, "ROS Time jumped backwards by %.3f s. Publishing costmaps anyway.", (prev_.toSec()-now.toSec()));
+      if (now < prev_ || last_publish_ + publish_cycle < now)
```

Comment:
Is this `prev_` variable needed? You could just do `if (now < last_publish_ || last_publish_ + publish_cycle < now)`, right?

Reply:
I though because time could jump in between cycles, but yeah that doesn't matter, so I changed it!